auth/internal/service/user.go

```diff
@@ -0,0 +1,35 @@
+package service
+
+import (
+	"context"
+
+	"auth/internal/model"
+	"auth/internal/repository"
+	"auth/internal/util"
+)
+
+type UserService interface {
+	CreateUser(ctx context.Context, email, password string) (*model.User, error)
+}
+
+type userService struct {
+	userRepository repository.UserRepository
+}
+
+func NewUserService(userRepository repository.UserRepository) UserService {
+	return &userService{
+		userRepository: userRepository,
+	}
+}
+
+func (s *userService) CreateUser(ctx context.Context, email, password string) (*model.User, error) {
+	hashedPassword, err := util.HashPassword(password)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.userRepository.CreateUser(ctx, &model.User{
+		Email:    email,
+		Password: hashedPassword,
+	})
+}
```

Comment:
Here you return to controller data directly from repository layer. it is not really bad, but it could leads to layers inconsistent models usage.
Think about it like DI: controller(service(repository)) -controller prepared data for service when calling it, service prepared data for repository. Then after receiving data from db repository should return data to service, service should get rid of database specific data and return it to controller, controller should prepare this data for external usage.